### ISSUE TITLE
Custom theme applies after ~2s after Zed start

### ISSUE THREAD

**Issue by widersky** [zed-industries/zed#8474]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When I start Zed, for the first second or two I see the default One Light theme, after a while the theme changes to my preferred one.

### Environment

Zed: v0.124.6 (Zed Preview)
OS: macOS 14.3.1
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by maxdeviant** [comment]
@widersky Is this a theme being provided via `~/.config/zed/themes` or from an extension?

**Comment by widersky** [comment]
@maxdeviant extension

**Comment by widersky** [comment]
@maxdeviant 16, mostly themes:

```
macOS Classic Theme
The Dark Side
Catppuccin Themes
GitHub Theme
Tokyo Night Themes
Zedokai
VSCode Dark Plus Theme
LOG
Smooth Theme
VSCode Monokai Charcoal
Mau Theme
Napalm Theme
base16
NeoSolarized
Xcode Themes
Horizon
```

**Comment by Congyuwang** [comment]
Same problem. Only extension themes have this problem.

For example, "macOS Classic theme":

https://github.com/zed-industries/zed/assets/52687642/18ccfa98-261e-4b08-9bc9-7d078592e965

**Comment by izuzak** [comment]
I've observed this as well. I have lots of extensions installed (51) since I like testing out other themes.

**Comment by Congyuwang** [comment]
> Same problem. Only extension themes have this problem.
> 
> For example, "macOS Classic theme":
> 
>  2024-03-07.12.09.39.mov

Same delay loading theme though I only have one extension installed.

**Comment by maxdeviant** [comment]
It looks like this might already be fixed (or at least, noticeably better) on `main` / Nightly after some changes we've made to extension loading.

Be on the lookout for improvements in Zed Preview v0.127.x when that ships on Wednesday.

**Comment by Congyuwang** [comment]
Looks like fixed in Zed Preview v0.127.0.

**Comment by Moshyfawn** [comment]
I'm going to close this one as the supposed fix has been released. Feel free to let me know if it's still happening.
